pkg/ctxlock/watcher.go

```diff
@@ -0,0 +1,35 @@
+package ctxlock
+
+// A Watcher waits on two cancellation sources and makes sure to call the
+// wrapped function as soon as possible.
+type watcher struct {
```

Comment:
Is there a reason behind the private struct with public methods?

Reply:
not particularly